clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -503,12 +503,16 @@ void Sema::checkSYCLDeviceVarDecl(VarDecl *Var) {
   checkSYCLType(*this, Ty, Loc, Visited);
 }
 
-// Tests whether given function is a lambda function or '()' operator used as
-// SYCL kernel body function (e.g. in parallel_for).
+// Tests whether OperatorFD is a lambda function or '()' operator used as
+// SYCL kernel body function (e.g. in parallel_for). If OperatorFD is a SYCL
+// kernel body function, it's parent (or parent's parent for wrapped kernel),
```

Comment:
I find "parent" word here is a bit confusing. There is a `getParent` function that can be called for each `DeclContext` including a function, and it returns a declaration context that is around. For functors it returns the class that defines `operator()`. If I understand correctly, here as "parent" you mean the function that calls the SYCL kernel body function. Maybe something like "caller" will work better.